src/app/game.ts

```diff
@@ -448,6 +449,172 @@ export class Game {
     this.storageCapacity += parseInt((warehouse.capacity * 1.5).toFixed(0))
   }
 
+  private checkIfRandomEvent() {
+    const randomInt = Math.floor(Math.random() * 10000)
+
+    if (randomInt < 30) {
+      this.handleRandomEvent()
+    }
+  }
+
+  private handleRandomEvent() {
+    const randomInt = Math.floor(Math.random() * 8)
+
+    switch (randomInt) {
+      case 0:
+        this.handleRandomGoldStealEvent()
+        break
+      case 1:
+        this.handleRandomBuilidngLevelDecreaseEvent()
+        break
+      case 2:
+        this.handleRandomWoodIncreaseEvent()
+        break
+      case 3:
+        this.handleMurderEvent()
+        break
+      case 4:
+        this.handleBornEvent()
+      case 5:
+        this.handleRandomTradeEvent()
+      case 6:
+        this.handleRandomStoneIncreaseEvent()
+      case 7:
+        this.handleRandomDefenceIncreaseEvent()
+      default:
+        break
+    }
+  }
+
+  private handleRandomGoldStealEvent() {
+    if (this.getGoldAmount() === 0) return
+
+    const goldTaken = Math.floor(0.05 * this.getGoldAmount())
+    if (goldTaken < 1) return
+
+    const gold = Math.floor(0.95 * this.getGoldAmount())
+    this.changeGoldAmount(gold)
+    this.addGameMessage({
+      message: `Oh no, thieves have stolen ${goldTaken} gold from your storage`,
+      type: MessageType.ERROR,
+    })
+  }
+
+  private handleRandomBuilidngLevelDecreaseEvent() {
+    const randomBuilding = this.buildings[
+      Math.floor(Math.random() * this.buildings.length - 1)
+    ]
+
+    if (!this.getBuilding(`${randomBuilding}`)) return
+
+    randomBuilding.setLevel(randomBuilding.getLevel() - 1)
+    this.addGameMessage({
+      message: `Watch out, there was a storm and lighting has hit your ${randomBuilding.getTitle()}. It's level got decreased by one.`,
+      type: MessageType.ERROR,
+    })
+  }
+
+  private handleMurderEvent() {
+    this.population = this.getPopulation() - 1
+    this.addGameMessage({
+      message:
+        'There was a murder in your village and population decreased by 1',
+      type: MessageType.ERROR,
+    })
+  }
+
+  private handleBornEvent() {
+    this.population++
+    this.addGameMessage({
+      message:
+        'New child was born in your village and your population increased by 1',
+      type: MessageType.SUCCESS,
+    })
+  }
+
+  private handleRandomWoodIncreaseEvent() {
+    if (!this.getBuilding('Sawmill')) {
```

Comment:
You can do it that:
const sawmill = this.getBuilding(Sawmill.name)
if (!sawmill) return;
...